signer/src/storage/postgres.rs

```diff
@@ -2121,6 +2152,50 @@ impl super::DbRead for PgStore {
         .map_err(Error::SqlxQuery)
     }
 
+    async fn is_withdrawal_active(
+        &self,
+        id: &model::QualifiedRequestId,
+        bitcoin_chain_tip: &model::BitcoinBlockRef,
+        min_confirmations: u64,
+    ) -> Result<bool, Error> {
+        let min_block_height = sqlx::query_scalar::<_, Option<i64>>(
+            r#"
+            SELECT MAX(bb.block_height) + 1
+            FROM sbtc_signer.bitcoin_withdrawals_outputs AS bwo
+            JOIN sbtc_signer.bitcoin_blocks AS bb
+              ON bb.block_hash = bwo.bitcoin_chain_tip
+            WHERE bwo.request_id = $1
+              AND bwo.stacks_block_hash = $2
+            "#,
+        )
+        .bind(i64::try_from(id.request_id).map_err(Error::ConversionDatabaseInt)?)
+        .bind(id.block_hash)
+        .fetch_one(&self.0)
+        .await
+        .map_err(Error::SqlxQuery)?;
+
+        // This means that there are no rows associated with the ID in the
+        // `bitcoin_withdrawals_outputs` table.
+        let Some(min_block_height) = min_block_height else {
+            return Ok(false);
+        };
+
+        let chain_tip_hash = &bitcoin_chain_tip.block_hash;
+        let least_txo_height = self
+            .get_least_txo_height(chain_tip_hash, min_block_height)
+            .await?
+            .map(u64::try_from)
+            .transpose()
+            .map_err(|_| Error::TypeConversion)?;
+        // If this returns None, then the sweep itself could be in the
+        // mempool. If that's the case then this is definitely active.
+        let Some(least_txo_height) = least_txo_height else {
+            return Ok(true);
+        };
+
+        Ok(least_txo_height + min_confirmations >= bitcoin_chain_tip.block_height)
```

Comment:
I found this a little easier to reason with 🤷 

```suggestion
        // We test whether the oldest UTXO height has fallen outside of the minimum
        // confirmation window.
        let max_allowed_height = bitcoin_chain_tip.block_height - min_confirmations;
        OK(least_txo_height <= max_allowed_height)
```

Reply:
Updated in https://github.com/stacks-network/sbtc/pull/1411/commits/3b8bfbaaee84886e4a2433d635a55b6b8714f2ad